bin/collator/src/parachain/service.rs

```diff
@@ -62,11 +62,20 @@ use crate::{
 };
 
 /// Parachain host functions
+#[cfg(not(feature = "runtime-benchmarks"))]
 pub type HostFunctions = (
     cumulus_client_service::ParachainHostFunctions,
     moonbeam_primitives_ext::moonbeam_ext::HostFunctions,
 );
 
+/// Host functions required for kitchensink runtime and Substrate node.
```

Comment:
we don't need this. The benchmark host functions is injected by benchmarking-cli itself

Reply:
I am getting this error if I remove benchmarking host functions:

```
Finished release [optimized] target(s) in 3m 14s
     Running `target/release/astar-collator benchmark overhead --dev --header=./.github/license-check/headers/HEADER-GNUv3`
Error: Service(Client(VersionInvalid("Other error happened while constructing the runtime: runtime requires function imports which are not present on the host: 'env:ext_benchmarking_proof_size_version_1', 'env:ext_benchmarking_add_to_whitelist_version_1', 'env:ext_benchmarking_current_time_version_1', 'env:ext_benchmarking_commit_db_version_1', 'env:ext_benchmarking_read_write_count_version_1', 'env:ext_benchmarking_wipe_db_version_1', 'env:ext_benchmarking_get_read_and_written_keys_version_1', 'env:ext_benchmarking_reset_read_write_count_version_1', 'env:ext_benchmarking_set_whitelist_version_1'")))
2024-12-02 13:52:50 Cannot create a runtime error=Other("runtime requires function imports which are not present on the host: 'env:ext_benchmarking_proof_size_version_1', 'env:ext_benchmarking_add_to_whitelist_version_1', 'env:ext_benchmarking_current_time_version_1', 'env:ext_benchmarking_commit_db_version_1', 'env:ext_benchmarking_read_write_count_version_1', 'env:ext_benchmarking_wipe_db_version_1', 'env:ext_benchmarking_get_read_and_written_keys_version_1', 'env:ext_benchmarking_reset_read_write_count_version_1', 'env:ext_benchmarking_set_whitelist_version_1'")
```

PS: I will fix the comment (copied)

Reply:
https://github.com/paritytech/polkadot-sdk/blob/c0921339f9d486981b3681760ee83ba9237f2eaa/substrate/utils/frame/benchmarking-cli/src/pallet/command.rs#L66
Not sure why you're getting this error

Reply:
seems like overhead doesn't include it, wonder why

Reply:
We can make an issue or PR for it, probably it's small effort.